cast/src/lib.rs

```diff
@@ -447,8 +453,40 @@ impl SimpleCast {
         format!("0x{}", s)
     }
 
+    pub async fn generate_interface(
+        address_or_path: String,
+        chain: Chain,
+        etherscan_api_key: Option<String>,
```

Comment:
Given that we want to support local paths now, it makes more sense to make the argument like below:

```
enum InterfacePath {
      Local(String)
      Etherscan { chain: Chain, api_key: String }
}
```

and make the signature `fn generate_interface(args: InterfacePath)`